src/common/android/deviceResolver.ts

```diff
@@ -0,0 +1,37 @@
+import {ChildProcess} from "../node/childProcess";
+import * as Q from "q";
+
+export interface IDevice {
+    id: string;
+    status: string;
+}
+
+export class DeviceResolver {
+
+    /**
+     * Gets the list of Android connected devices and emulators.
+     */
+    public getConnectedDevices(): Q.Promise<IDevice[]> {
```

Comment:
I disagree. The adb parsing logic here is done with a regex in 10 lines of code. The rest of this PR is business logic. I don't think we should go take a new dependency for this much code.